addons/editor-delete-button/userscript.js

```diff
@@ -0,0 +1,56 @@
+export default async ({ addon, msg }) => {
+  const { redux } = addon.tab;
+  // Wait until user has logged in, and is the author of the project
+  await redux.waitForState((state) => state.preview?.projectInfo?.author?.id === state.session?.session?.user?.id);
+  while (true) {
+    const loadItem = await addon.tab.waitForElement(
+      "div[class^='menu-bar_file-group'] > :nth-child(3) ul > :nth-child(4):not(.sa-editor-delete-button)",
+      {
+        markAsSeen: true,
+        reduxCondition: (state) => !state.scratchGui.mode.isPlayerOnly && !state.preview.visibilityInfo.deleted,
+      }
+    );
+
+    const dropdownItem = document.createElement("li");
+    dropdownItem.className = addon.tab.scratchClass("menu_menu-item", "menu_hoverable", {
+      others: ["sa-editor-delete-button"],
+    });
+    dropdownItem.innerText = msg("button-text");
+    addon.tab.displayNoneWhileDisabled(dropdownItem, { display: "block" });
+
+    dropdownItem.addEventListener("click", async (e) => {
+      const confirmed = await addon.tab.confirm(msg("modal-title"), msg("modal-msg"), {
+        okButtonLabel: msg("yes"),
+        cancelButtonLabel: msg("no"),
+        useEditorClasses: true,
+      });
+      if (!confirmed) return;
+      const res = await fetch(`/site-api/projects/all/${redux.state.preview.projectInfo.id}/`, {
+        headers: {
+          "X-CSRFToken": addon.auth.csrfToken,
+        },
+        method: "PUT",
+        body: JSON.stringify({
+          visibility: "trshbyusr",
+        }),
+      });
+
+      if (res.ok) {
+        redux.dispatch({
+          type: "SET_VISIBILITY_INFO",
+          visibilityInfo: {
+            ...redux.state.preview.visibilityInfo,
+            deleted: true,
+          },
+        });
+        // TODO: Maybe go to mystuff instead?
+        redux.dispatch({
+          type: "scratch-gui/mode/SET_PLAYER",
+          isPlayerOnly: true,
+        });
+      }
+    });
+
+    loadItem.insertAdjacentElement("beforebegin", dropdownItem);
```

Comment:
A better position for this option would be at the bottom of the menu, with a separator above:
![image](https://user-images.githubusercontent.com/51849865/177110623-153fe804-71d1-48b1-8666-7e6d4300c891.png)